distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -1022,3 +1033,95 @@ Text.to_case case_option=Case.Lower locale=Locale.Default = case case_option of
     Case.Lower -> UCharacter.toLowerCase locale.java_locale this
     Case.Upper -> UCharacter.toUpperCase locale.java_locale this
     Case.Title -> UCharacter.toTitleCase locale.java_locale this Nothing
+
+## Returns the input padded to the specified `length`, using the `with_pad`
+   string repeated at the start or the end.
+
+   Arguments:
+   - length: The new length for the output. The result is the original string if
+     the input length is more than length.
+   - with_pad: The string to use to pad the input. If the last repetition
+     exceeds the target length, it is truncated to the required size. If padding
+     at the `End`, the beginning of the padding string is used and if padding at
+     `Start`, the end of the string is used.
+   - at: The location of where to pad the input.
+
+   > Example
+     Padding a text with whitespace at the end.
+
+         "Hello World!".pad 15 == "Hello World!   "
+
+   > Example
+     Behavior of padding if the `with_pad` string has to be truncated.
+
+         "HELLO".pad 9 "AB" == "HELLOABAB"
+         "HELLO".pad 8 "AB" == "HELLOABA"
+         "HELLO".pad 8 "AB" Start == "BABHELLO"
+
+Text.pad : Integer -> Text -> (Location.Start | Location.End) -> Text
+Text.pad length=0 with_pad=' ' at=Location.End =
+    with_pad_length = with_pad.length
+    if with_pad_length == 0 then Error.throw (Illegal_Argument_Error "`with_pad` must not be an empty string.") else
+        pad_size = length - this.length
+        if pad_size <= 0 then this else
+            full_repetitions = pad_size.div with_pad_length
+            remainder = pad_size % with_pad_length
+            case at of
+                Location.Start ->
+                    with_pad.take (Text_Sub_Range.Last remainder) + with_pad.repeat full_repetitions + this
```

Comment:
is it worth short-circuiting if remainder == 0? Especially as Last would involve a scan of the with_pad?

Reply:
Made Last avoid the scan.